### PR TITLE
refactored `resolve` method logic to not use `inet_pton`

### PR DESCRIPTION
**Pull request by austinheap**
`inet_pton` throws warnings -- which while prior to this PR were muted with by `@` -- are avoidable.

### CODE REVIEW

**Comment by kelunik** [comment]
What's the benefit here?

**Comment by austinheap** [comment]
@kelunik Suppressing an error message with `@` doesn't mean the error didn't occur, just that the PHP runtime won't display a message about it. Custom error handler (set with `set_error_handler`) are still executed and still views it as an error though, so if you're using custom error handlers (say `xdebug`, `sentry`, etc) this single line of code gets super loud with errors.

It didn't seem to make sense to trigger an apps custom error handlers just to detect if an IP was used as an argument and should be returned immediately as a record as long as the type restriction matched.

TLDR; This PR accomplishes the same thing, but without suppressed errors which other packages still view as actual errors.

(Thanks for @amphp btw!)

**Comment by kelunik** [comment]
@austinheap The thing is, PHP is stupid. It outputs errors where it shouldn't. If your custom error handler doesn't handle suppressed errors correctly, you'll get other unwanted error messages, because e.g. streams with a full buffer will result in warnings, even if the code is correct and there isn't anything that can be improved.

I agree, however, that there might be a use case for an error handler that doesn't obey the error suppression, because it might be used to hunt down bugs where too many errors are suppressed.

@trowski Thoughts?

**Comment by austinheap** [comment]
@kelunik Totally agree re: PHP being stupid with warnings like this and from streams, like you mention. A project I'm working on does *lots* of queries using this package and the xdebug output isn't useable due to the amount of warning spam related to this suppressed `inet_pton` call.

**Comment by DaveRandom** [comment]
Can't we just fix this in php-src?

The errors emitted by `inet_pton()`/`inet_ntop()` are entirely useless, to the extent that it might even be permissible to merge this down to 7.1 (I will ask Joe).

**Comment by kelunik** [comment]
That'd be my preference here, too. Please, do.

**Comment by austinheap** [comment]
@DaveRandom super cool idea, thanks so much for pushing this!

**Comment by kelunik** [comment]
@trowski I guess we can do that, yes.

@austinheap Please add `ext-filter` to the `composer.json` requirements.

**Comment by austinheap** [comment]
@kelunik Done.

**Comment by kelunik** [comment]
Thanks!
